src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -1949,7 +1937,58 @@ protected override bool TryGetField(FormulaType fieldType, string fieldName, out
             "f():TestEntity = Entity; g(e: TestEntity):Number = 1;",
             "g(f())",
             true,
-            1.0)]
+            1.0)]
+
+        // Aggregate types with more than expected fields are not allowed in UDF args and return types
+        // Records
+        [InlineData(
+            "f():T = {x: 5, y: 5}; T := Type({x: Number});",
+            "",
+            false)]

```

Comment:
>false [](http://example.com/codeflow?start=12&length=5)

Can we have the tests check whether the error comes from the UDF definition or from its usage? Currently we just check that there is an error in any of those.